### ISSUE TITLE
Update to the latest mintty code... not exactly a request...

### ISSUE THREAD

**Issue by paolo-sz** [juho-p/fatty#26]
Hi,

this the only way I found on GitHub to get in touch with you...
I notice the Fatty code is based on a very old version of mintty source code so I took some time to try to realign to the most recent version.
You can have a look of what I have done on https://github.com/paolo-sz/fatty.
The code compiles and Fatty does not suddenly crash just after starting but I'm 200% sure the re-alignment is full of hidden issues (maybe big issues).
The changes are pretty wide and I do not know if they are of any interest to you so I have chose to not directly open a pull request.
Before starting the mintty code realignment I did some commits in order to fix (or try to fix) a couple of issues of your current code (e804395162f0172885a63566172f557593f24be7, f3664a05edeb4888ffb0074d9b3b04157ab3dd6a, 332b9ce4caee21c5535d13c8acf8d1881126a2ee, 88334226b8eb12efa1e06e7f6419d577588f720d). These single commits could be merged into your tree without too much issues.
Also I made an additional improvement using the TabCtrl window provided by Microsoft in order to manage the tabs (832b2fcb8c43003a627cac5199d7ea3e0651310c). Also this commit could be merged into your tree without too much issues.
In all the other commits (mintty alignment), most of the changes are "simply" a fix due to your changes on term and child structures (pointers to) but there are also big features additions like searchbar (4955e5dc751a6bd1b9330550404a8af439bbce2a).
Also, along the way, mintty started using Shift+LeftArrow/RightArrow so I changed the tab swap key combination in Ctrl+Shift+PgUp/PgDn and the previous/next tab key combination in Ctrl+PgUp/PgDn (which, by the way, are the standard key combination for the same feature on linux gnome terminal).
Let me know if you are interested in any of the commit I have done.

**Comment by juho-p** [comment]
This looks interesting. It takes some time (slightly busy at the moment) for me to see what you have actually done before I can merge (or cherry-pick) these. At least those fixes will probably get merged in near future.

**Comment by paolo-sz** [comment]
That's fine. It took me pretty a lot of time just to, more or less, blindly merge all the  commit from mintty tree so I understand that putting some more brain in the work can take a lot of time. In the meanwhile I will use this version and I will try to fix any misbehavior I will find (I have already found some little ones).
By the way, I have really a bad feeling with git and "she" probably has a bad feeling with me too so be prepared to some damned push made wrong... sigh...

**Comment by paolo-sz** [comment]
I just started to find issues on my changes prior to mintty update. I made a fix but this could became a mess for you when you will start the review so I created a branch (fixes2original) where all the new features are not included.

**Comment by juho-p** [comment]
Thanks for your fixes. Sorry it took so long for me to look into this. I merged your fixes2original branch. I didn't have time to test it very well though so I hope it works out well... I commented the code in your repository but those were just some ideas for you to consider and not that bad problems.

I'm still considering if I can merge your master-branch.

**Comment by paolo-sz** [comment]
Really, do not worry at all, I understand people have something called "real life"... :-)

Take your time and do not feel obliged in any way.
